client.go

```diff
@@ -255,14 +255,14 @@ func FromRequest(r *http.Request) (*Request, error) {
 	return &Request{bodyReader, r}, nil
 }
 
-// NewRequest creates a new wrapped request.
-func NewRequest(method, url string, rawBody interface{}) (*Request, error) {
+// NewRequestWithContext creates a new wrapped request.
```

Comment:
I would go a little more descriptive here and mention what happens with the context.